engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -253,18 +262,64 @@ private boolean canHandle(DataObject dataObject) {
         return false;
     }
 
+    /**
+     * True if volumes source storage are NFS
+     */
+    protected boolean isSourceNfsPrimaryStorage(Map<VolumeInfo, DataStore> volumeMap) {
+        if (MapUtils.isNotEmpty(volumeMap)) {
+            for (VolumeInfo volumeInfo : volumeMap.keySet()) {
+                StoragePoolVO storagePoolVO = _storagePoolDao.findById(volumeInfo.getPoolId());
+                return storagePoolVO != null &&
+                        storagePoolVO.getPoolType() == Storage.StoragePoolType.NetworkFilesystem;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * True if destination storage is cluster-wide NFS
+     */
+    protected boolean isDestinationNfsPrimaryStorageClusterWide(Map<VolumeInfo, DataStore> volumeMap) {
```

Comment:
@nvazquez The volume info in the Map is unused. Better pass pools list as input here.

Reply:
Same as before, the consumer will have to check the map for nullity, unbox the map and pass the list.

Reply:
Understood. Either case nullity check has to be done. But, I think this is not a good coding practice, to pass the unused data to the method.

Reply:
Well depends how you'd want to look at it. It's not unused if you're using the collection.

Reply:
For a datastore validation, only datastore details are required.